### PR TITLE
service uuid based ble tracking

### LINKED ISSUE #94: iBeacon tracking
**Describe the problem you have/What new integration you would like**
BLE tracking - can't we use the UUID in esphome yaml files to track phones?

**Please describe your use case for this integration and alternatives you've tried:**
I have an app on my Android phone that mimics an iBeacon and emits a constant UUID. Can't this be used to do room based presence detection.

**Additional context**
In BeaconScope (an Android app) I can emit different types of beacons and configure the uuid. Can I use esphome to track this iBeacon.

### PR DESCRIPTION
**Pull request by Lumpusz**
## Description:
Allow service uuid based ble tracking to enable tracking android phones despite the mac randomization.

I tested the svc uuids with nrfConnect.

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/94

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>
https://github.com/esphome/esphome-docs/pull/377

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/ble_rssi/ble_rssi_sensor.h [hunk 1]
```diff
@@ -11,17 +11,67 @@ namespace ble_rssi {
...
-      return true;
+    if (this->by_address_) {
+      if (device.address_uint64() == this->address_) {
+        this->publish_state(device.get_rssi())  ;
+        this->found_ = true;
+        return true;
+      }
+    } else {
+      for (auto uuid : device.get_service_uuids()) {
+	switch (this->uuid_.get_uuid().len) {
```
Comment (OttoWinter) [inline esphome/components/ble_rssi/ble_rssi_sensor.h]: Miked tag and space formatting - esphome uses spaces exclusively (see also CI)

**Review by OttoWinter — APPROVED** [review]

**Comment by Lumpusz** [comment]
Which version will you include this PR in?

**Comment by OttoWinter** [comment]
1.14 release has been cut - so this will be in 1.15

One goal for the next release will be to make it easier to load custom components when something's not yet released - for now, I'd recommend using the `custom_components` folder (see contributing guide)

**Comment by danibjor** [comment]
> One goal for the next release will be to make it easier to load custom components when something's not yet released

Interesting. Care to elaborate?

**Comment by OttoWinter** [comment]
@danibjor Nothing's implemented yet, but see also https://github.com/esphome/feature-requests/issues/314

**Pull request merged**